feat(core_kit): implement AppSectionHeader component with Widgetbook stories

mrezan12 opened the pull request:
# Pull Request

## 📄 Summary

Implement the AppSectionHeader component for organizing content into logical groups with titles and optional actions. This widget is commonly used in settings screens, forms, lists, and navigation drawers.

**Features:**
- Title and subtitle support
- Leading and trailing widget support
- Divider support with position control (above, below, both)
- Material Design 3 specifications (48dp min height, proper typography and colors)
- Accessibility with semantic header role
- SectionHeaderDelegate for sticky header support

---

## 🧩 Affected Module(s)

- [x] Packages (packages/*)
- [x] Widgetbook Kit (widgetbook_kit/)
- [ ] Documentation (docs/)
- [ ] CI / Infra (.github/)

---

## ✅ Checklist

- [x] My branch name follows format: <type>/<short-description> (e.g., feat/login-screen)
- [x] My PR title starts with one of the approved types listed above
- [x] My Dart code is formatted (dart format . or flutter format .)
- [x] I ran static analysis (flutter analyze) and resolved warnings
- [x] I ran tests successfully (flutter test)
- [x] I updated / checked pubspec.yaml and pubspec.lock for dependency changes
- [x] For UI changes, I added screenshots and/or updated golden tests (if applicable)
- [x] I linked related issues using keywords like Closes #42
- [x] I ensured this PR has no unrelated changes
- [x] This PR is ready for review and does not include unfinished work

---

## 🔗 Related Issues

Closes #20

---

## 💬 Additional Notes (Optional)

### Files Changed:
- [packages/core_kit/lib/widgets/surfaces/app_section_header.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/lib/widgets/surfaces/app_section_header.dart:0:0-0:0) - Widget implementation
- [packages/core_kit/lib/core_kit.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/lib/core_kit.dart:0:0-0:0) - Export added
- [packages/core_kit/test/widgets/surfaces/app_section_header_test.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/packages/core_kit/test/widgets/surfaces/app_section_header_test.dart:0:0-0:0) - 27 widget tests
- [widgetbook_kit/lib/stories/core_kit/widgets/surfaces/app_section_header_stories.dart](cci:7://file:///home/mrcan/Documents/projects/hexaMobileShare/widgetbook_kit/lib/stories/core_kit/widgets/surfaces/app_section_header_stories.dart:0:0-0:0) - 8 Widgetbook stories

### Widgetbook Stories:
1. Default
2. With Subtitle
3. With Leading
4. With Trailing
5. With Divider
6. Settings List Example
7. Theme Variations
8. Interactive Playground

mrezan12 committed:
feat(core_kit): implement AppSectionHeader component with Widgetbook stories

- Add AppSectionHeader widget with title, subtitle, leading, trailing support
- Implement DividerPosition enum for flexible divider placement
- Add dividerIndent and dividerEndIndent for divider customization
- Implement SectionHeaderDelegate for sticky header support
- Follow Material Design 3 specs (48dp min height, proper typography)
- Add accessibility support with semantic header role
- Add comprehensive widget tests (27 tests)
- Create 8 Widgetbook stories for component demonstration

Closes #20

husamettinarabaci reviewed (changes requested):
## Review Summary

Thank you for the comprehensive implementation of the AppSectionHeader component! The PR is in excellent shape with complete test coverage (27 tests) and comprehensive Widgetbook stories (8 stories). All CI checks are passing.

However, there are a few minor issues that need to be addressed before merging:

---

## ⚠️ Required Changes

### 1. **Semantic Label Accessibility Issue** (MUST FIX)

**File**: `packages/core_kit/lib/widgets/surfaces/app_section_header.dart:255`

**Issue**: The semantic label uses `title` instead of `displayTitle`, which causes a mismatch between what screen readers announce and what users see when `isUpperCase=true`.

**Current code**:
```dart
label: subtitle != null ? '$title, $subtitle' : title,
```

**Required fix**:
```dart
label: subtitle != null ? '$displayTitle, $subtitle' : displayTitle,
```

**Why this matters**: Accessibility compliance requires that screen readers announce exactly what's displayed visually. When `isUpperCase=true`, the visual display shows uppercase text, but screen readers would announce the original case, creating confusion for users with visual impairments.

---

## 📝 Recommended Changes (Best Practice)

### 2. **Redundant Uppercase in Widgetbook Stories**

**File**: `widgetbook_kit/lib/stories/core_kit/widgets/surfaces/app_section_header_stories.dart`
**Lines**: 162, 180, 198

**Issue**: Titles are already provided in uppercase AND `isUpperCase: true` is also set. This is redundant and doesn't demonstrate the correct usage pattern.

**Current code**:
```dart
const AppSectionHeader(title: 'ACCOUNT', isUpperCase: true),
const AppSectionHeader(title: 'PRIVACY & SECURITY', isUpperCase: true),
const AppSectionHeader(title: 'NOTIFICATIONS', isUpperCase: true),
```

**Recommended fix**:
```dart
const AppSectionHeader(title: 'Account', isUpperCase: true),
const AppSectionHeader(title: 'Privacy & Security', isUpperCase: true),
const AppSectionHeader(title: 'Notifications', isUpperCase: true),
```

**Why this matters**: The Widgetbook stories serve as usage examples for other developers. They should demonstrate the correct pattern: provide normal case titles and let the component handle the transformation when `isUpperCase` is enabled.

---

## ✅ What's Excellent About This PR

- ✅ Complete implementation matching all requirements from Issue #20
- ✅ Material Design 3 compliance (48dp min height, proper typography, colors)
- ✅ Comprehensive test coverage (27 tests covering all scenarios)
- ✅ Excellent Widgetbook stories (8 stories with interactive playground)
- ✅ Proper accessibility with semantic header role
- ✅ Sticky header support via SectionHeaderDelegate
- ✅ Clean, well-documented code with Dartdoc comments
- ✅ All CI checks passing (format, analyze, tests, REUSE compliance)
- ✅ SPDX headers present in all files
- ✅ Proper export in core_kit.dart

---

## 📋 Action Items

Please address:
1. **[REQUIRED]** Fix the semantic label to use `displayTitle` instead of `title`
2. **[RECOMMENDED]** Update Widgetbook story titles to normal case (let `isUpperCase` flag handle transformation)

Once these changes are made, this PR will be ready to merge! 🚀

---

**Estimated effort**: ~5 minutes to fix both issues.

Great work on this implementation! 👏

mrezan12 committed:
fix(core_kit): address PR review feedback for AppSectionHeader

- Fix accessibility: use displayTitle in semantic label for correct screen reader output
- Update Widgetbook story titles to normal case (let isUpperCase handle transformation)

mrezan12 committed:
feat(widgetbook): make switches interactive in Settings List Example

- Convert Settings List Example to StatefulWidget for realistic usage
- Switches now maintain state when toggled

husamettinarabaci reviewed (dismissed):
LGTM

husamettinarabaci committed:
Merge branch 'develop' into feat/app-section-header
